rules/TypeDeclaration/NodeAnalyzer/ReturnTypeAnalyzer/StrictScalarReturnTypeAnalyzer.php

```diff
@@ -47,4 +57,25 @@ public function matchAlwaysScalarReturnType(ClassMethod|Closure|Function_ $funct
 
         return $this->typeFactory->createMixedPassedOrUnionType($scalarTypes);
     }
+
+    private function isHardCodedExpression(Expr $expr): bool
+    {
+        // Normal scalar values like strings, integers and floats
+        if ($expr instanceof Scalar) {
```

Comment:
Didn't notice this before, but apparently the magic constants extend `Scalar` as well. So the magic constants like `__METHOD__` etc will be considered to be "hard coded" by this logic. Personally I'm fine with that, but it might be unexpected.

Possible solution would be:
```suggestion
        if ($expr instanceof Scalar && ! $expr instanceof MagicConst) {
```
(and adding `use PhpParser\Node\Scalar\MagicConst` obviously)

Reply:
seems ok 👍

Reply:
Once more, I kept it as is then 😄 (so magic constants are threaded as "hard coded")